src/04-modality-specific-files/10-microscopy.md

```diff
@@ -0,0 +1,396 @@
+# Microscopy
+
+Support for Microscopy was developed as a
+[BIDS Extension Proposal](../07-extensions.md#bids-extension-proposals).
+
+Please see [Citing BIDS](../01-introduction.md#citing-bids)
+on how to appropriately credit this extension when referring to it in the
+context of the academic literature.
+
+<!---
+The following link is temporary and points to the bep031 branch of bids-examples,
+update to 'https://github.com/bids-standard/bids-examples' when examples are merged.
+-->
+Microscopy datasets formatted using this specification are available on the
+[BIDS examples repository](https://github.com/neuropoly/bids-examples/tree/bep031)
+(`microscopy_SEM001` and `microscopy_SPIM001`) and can be used for practical
+guidance when curating a new dataset.
+
+Further Microscopy datasets are available:
+
+-   In PNG format:  [`data_axondeepseg_sem`](https://doi.org/10.5281/zenodo.5498378)
+-   In OME-TIFF format: [`Broca's Area Light-Sheet Microscopy`](https://doi.org/10.5281/zenodo.5517223)
+
+## Microscopy recording data
+<!---
+The following table will be generated automatically with macros after community review.
+A "manual" table is provided to facilitate the review process.
+-->
+Template:
+```Text
+sub-<label>/
+    [ses-<label>/]
+        microscopy/
+            sub-<label>[_ses-<label>]_sample-<label>[_acq-<label>][_stain-<label>][_run-<index>][_chunk-<index>]_<suffix>.<extension>
```

Comment:
For my current dataset, I'm restricting my NGFF instances to single channels, even though the data is inherently multichannel. The naming convention and the structure of the sidecars only let you represent a single channel / stain with a volume. One of your NGFF indices is always zero (the channel index) but it seems necessary to make it fit into the larger BIDS scheme. It's necessary to split the channels among several files.

Reply:
> A single OME-TIFF could conceivably contain multiple chunks/runs/etc. (A situation which might become more pronounced with NGFF) Is there a provision for linking to them, or would a user need to split the file(s)?
> 
> Edit: Ah. I assume due to "to accommodate datasets stored in 2D image formats and whole-slide imaging formats" that the intent is more to avoid this situation.

@joshmoore The `chunk`, `run` and `stain` entity are optional (between `[]`), which means you can use them if you need them to distinguish between files that otherwise would have the same name.
If all the chunks are in the same file, you may not used them and it is not require to split into separate files.
If the chunks are in separate files, the `chunk` entity should be used.

@LeeKamentsky Similarly, there is a way in the current specification to describe multiple stains (one per channel). It is described in the SampleStaining field [here](https://bids-specification--881.org.readthedocs.build/en/881/04-modality-specific-files/10-microscopy.html#sample). So in that case (multi-channels/multi-stains), what you can do is omit the `stain` entity in the filename and describe the stainings as a list in the metadata. This example is also described in the `satin-<label>` part [here](https://bids-specification--881.org.readthedocs.build/en/881/04-modality-specific-files/10-microscopy.html#filename-entities).

Reply:
Hi @LeeKamentsky, I also wanted to mention that NGFF is not officially supported by this BEP so far. As discussed in our meetings, we included it as a future format while still in development.

Reply:
Thanks @mariehbourget . For our current stain set, the YO and LEC channels are chemical stains and the NeuN stain is an anitbody one. I'm guessing I could do something like:
```
{
  "SampleStaining":[
    "YOYO",
    "NN",
    "LEC"],
  "SamplePrimaryAntibody": [
    "n/a",
    "Rabbit` Polyclonal Antibody to FOX3/NeuN Cat# RPCA-FOX3",
    "n/a"],
  "SampleSecondaryAntibody": [
    "n/a",
    "Rhodamine Red\u2122-X (RRX) AffiniPure Donkey Anti-Rabbit IgG (H+L), 711-295-152",
    "n/a"]
}
```

Reply:
Thanks @LeeKamentsky, this is an interesting case.
I think your suggestion is the best way to do it (if you want to keep a single file).

Maybe we should change the text for the stain-label

from:
```
Stains or antibodies SHOULD be indicated as appropriate in the `"SampleStaining"`,
`"SamplePrimaryAntibodies"` and/or `"SampleSecondaryAntobodies"` keys in the sidecar JSON file,
although the label may be different.
```

to:
```
Stains SHOULD be indicated in the `"SampleStaining"` key in the sidecar JSON file,
although the label may be different. Description of antibodies SHOULD also be indicated in
`"SamplePrimaryAntibodies"` and/or `"SampleSecondaryAntobodies"` as appropriate.
```

Would that be clearer?

Reply:
That's clearer. Thanks.

Reply:
> @joshmoore The `chunk`, `run` and `stain` entity are optional (between `[]`), which means you can use them if you need them to distinguish between files that otherwise would have the same name.
> If all the chunks are in the same file, you may not used `chunk` and it is not require to split into separate files.
> If the chunks are in separate files, the `chunk` entity should be used.

Just to be clear, are you proposing that one can either have a multi-(chunk/stain/run) file or split into multiple (chunks/stains/runs)? I would caution that any flexibility in representation is a burden on recipients of the dataset, both researchers and tools.

Reply:
@effigies - with ngff this will be possible and we definitely want to have that possibility. see ongoing discussion here: https://github.com/ome/ngff/issues/74 (this is not a burden but an enhancement when you are dealing with single blobs that are TBs in size.

Reply:
as the field evolves, such raw chunks may eventually become like MRI k-space data and the only data that is stored is some stitched and resampled multi-resolution version. in fact that's already happening to some extent since the original data often has oblique acquisitions, which are not being stored, but resampled into these rectangular chunks.

Reply:
The burden I mean is to understand two disparate representations to be identical and process them accordingly. I did not mean to say it shouldn't be done, just that when we permit such flexibility, it's worth the cost. As long as microscopy folks are satisfied, I have no objections.

Reply:
@effigies, yes, we are proposing either solutions i.e. multi-(chunk/stain/run) or split into multiple files.
As we discussed with contributors in the developement of the BEP, it seems that both situations are common and can depend on the acquisition protocol.

For example with `stains`, sometimes the stains are acquired separately in separate files, but sometimes acquired together in the same file, in the same session. This is why we have both possibilities i.e. having `stain-label` in the filename or a list of stains in the `SampleStaining` metadata field.

Same goes for `chunks`, where multiple chunks can be in the same file or not. Moreover, `chunk-label` also covers 2D PNG cases or cases where an "old" dataset is converted to BIDS with multiple images per samples.

I personally had not thought of the `run` case before. I assume the same argument would stand, but this may be less in line with the current BIDS definition.

Reply:
@LeeKamentsky, I integrated the change in the stain-label text to clarify the usage of SampleStaining. Thank you!